pkg/engine/mutation.go

```diff
@@ -115,6 +121,8 @@ func Mutate(policyContext *PolicyContext) (resp *response.EngineResponse) {
 					incrementAppliedCount(resp)
 				}
 			}
+
+			policyContext.JSONContext.Reset()
```

Comment:
It will be great if  we can add the comment here, why we need to reset the JSON context.

Reply:
It breaks another test for chained rules. Working on the fix.